droidlet/interpreter/robot/tasks.py

```diff
@@ -359,6 +365,159 @@ def step(self):
+        print(f'CuriousExplore task_data {task_data}')
+        TaskNode(agent.memory, self.memid).update_task(task=self)
+
+    def init_curious_logger(self):
+        self.logger = logging.getLogger('curious')
+        self.logger.setLevel(logging.INFO)
+        fh = logging.FileHandler(f"curious_explore_{'_'.join([str(x) for x in self.goal])}.log", 'w')
+        fh.setLevel(logging.INFO)
+        ch = logging.StreamHandler()
+        ch.setLevel(logging.INFO)
+        formatter = logging.Formatter('%(filename)s:%(lineno)s - %(funcName)s(): %(message)s')
+        fh.setFormatter(formatter)
+        self.logger.addHandler(fh)
+        self.logger.addHandler(ch)
+        self.logger.info(f'CuriousExplore task_data {self.task_data}')
+
+    @Task.step_wrapper
+    def step(self):
+        super().step()
+        self.interrupted = False
+        self.finished = False
+        
+        if self.steps[0] == "not_started":
+            if self.agent.mover.nav.is_done_exploring().value:
+                # clear memory
+                objects = DetectedObjectNode.get_all(self.agent.memory)
+                self.logger.info(f'Beginning to clear {len(objects)} memids ...')
+                self.agent.memory.clear(objects)
+                ExaminedMap.clear()
+                # reset object id counter
+                self.agent.perception_modules["vision"].vision.deduplicate.object_id_counter = 1
+                objects = DetectedObjectNode.get_all(self.agent.memory)
+                self.logger.info(f'{len(objects)} present now.')   
+            print(f'exploring goal {self.goal}')           
+            self.agent.mover.explore(self.goal)
+            self.dbg_str = "Explore"
+            self.steps[0] = "finished"
+            return
+        
+        # execute a examine maneuver
+        if self.steps[0] == "finished" and self.steps[1] == "not_started":
+            # Get a list of current detections
+            objects = DetectedObjectNode.get_all(self.agent.memory)
+            pos = self.agent.mover.get_base_pos_in_canonical_coords()
+            # pick all from unexamined, in-sight object
+            def pick_random_in_sight(objects, base_pos):
+                for x in objects:
+                    if ExaminedMap.can_examine(x):
+                        # check for line of sight and if within a certain distance
+                        yaw_rad, _ = get_camera_angles([base_pos[0], CAMERA_HEIGHT, base_pos[1]], x['xyz'])
+                        # self.logger.info(f"{x['label']}, yaw_rad {yaw_rad}, base_pos {base_pos}")
+                        dist = np.linalg.norm(base_pos[:2]-[x['xyz'][0], x['xyz'][2]])
+                        if abs(yaw_rad - base_pos[2]) <= math.pi/4 and dist <= 3:
+                            self.logger.info(f"Exploring eid {x['eid']}, {x['label']} next, dist {dist}")
+                            return x
+                return None
+            target = pick_random_in_sight(objects, pos)
```

Comment:
should we have some logic to explore all detected objects, or otherwise encourage not looking at the same things over and over again?

Reply:
We do check to ensure that the same region isn't explored more than twice (https://github.com/facebookresearch/fairo/pull/859/files#diff-59137b4e93337879456a9581c814274ac572e6abba78c8f0bbb56b07deb2a12aR260). As for exploring all detected objects, it's something we can try as a follow-up.